File: app/src/main/java/sfllhkhan95/doodle/models/Thumbnail.java
package sfllhkhan95.doodle.models;

import android.graphics.Bitmap;

public class Thumbnail {
    private Bitmap icon;
    private String name;

    public Thumbnail(Bitmap icon, String name) {
        this.icon = icon;
        this.name = name;
    }

    public Bitmap getIcon() {
        return icon;
    }

    public void setIcon(Bitmap icon) {
        this.icon = icon;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: app/src/main/java/sfllhkhan95/doodle/projects/models/Thumbnail.kt
package sfllhkhan95.doodle.projects.models

import android.graphics.Bitmap
import android.view.View
import sfllhkhan95.doodle.R
import sfllhkhan95.doodle.projects.utils.ThumbnailInflater

class Thumbnail(
        private val inflater: ThumbnailInflater,
        var icon: Bitmap?,
        var name: String?) :
        View.OnClickListener {

    override fun onClick(view: View) {
        when (view.id) {
            R.id.shareButton -> name?.let { inflater.share(name!!) }
        }
    }

}